google/cloud/spanner/samples/samples.cc

```diff
@@ -3649,6 +3666,250 @@ void DropForeignKeyConstraintDeleteCascade(
+    google::cloud::spanner_admin::DatabaseAdminClient client,
+    std::string const& project_id, std::string const& instance_id,
+    std::string const& database_id) {
+  google::cloud::spanner::Database database(project_id, instance_id,
+                                            database_id);
+  google::spanner::admin::database::v1::UpdateDatabaseDdlRequest request;
+  request.set_database(database.FullName());
+  google::protobuf::FileDescriptorSet fds;
+  google::cloud::spanner::testing::SingerInfo::default_instance()
+      .GetMetadata()
+      .descriptor->file()
+      ->CopyTo(fds.add_file());
+  fds.SerializeToString(request.mutable_proto_descriptors());
+  request.add_statements(R"""(
+      CREATE PROTO BUNDLE (
+          google.cloud.spanner.testing.SingerInfo,
+          google.cloud.spanner.testing.Genre,
+      )
+  )""");
+  request.add_statements(R"""(
+      ALTER TABLE Singers ADD COLUMN
+          SingerInfo google.cloud.spanner.testing.SingerInfo
+  )""");
+  request.add_statements(R"""(
+      ALTER TABLE Singers ADD COLUMN
+          SingerInfoArray ARRAY<google.cloud.spanner.testing.SingerInfo>
+  )""");
+  request.add_statements(R"""(
+      ALTER TABLE Singers ADD COLUMN
+          SingerGenre google.cloud.spanner.testing.Genre
+  )""");
+  request.add_statements(R"""(
+      ALTER TABLE Singers ADD COLUMN
+          SingerGenreArray ARRAY<google.cloud.spanner.testing.Genre>
+  )""");
+  auto metadata = client.UpdateDatabaseDdl(request).get();
+  if (!metadata) throw std::move(metadata).status();
+  std::cout << "`Singers` table altered, new DDL:\n" << metadata->DebugString();
+}
+// [END spanner_add_proto_type_columns]
+
+// [START spanner_update_data_with_proto_message_column]
+void UpdateDataWithProtoMessageColumn(google::cloud::spanner::Client client) {
+  google::cloud::spanner::testing::SingerInfo singer_proto;
+  singer_proto.set_singer_id(2);
+  singer_proto.set_birth_date("1942-06-18");
+  singer_proto.set_nationality("British");
+  singer_proto.set_genre(google::cloud::spanner::testing::Genre::POP);
+
+  using SingerInfoMessage = google::cloud::spanner::ProtoMessage<
+      google::cloud::spanner::testing::SingerInfo>;
+  auto singer_info = SingerInfoMessage(singer_proto);
+  auto commit_results =
+      client.CommitAtLeastOnce({google::cloud::spanner::Mutations{
+          google::cloud::spanner::InsertOrUpdateMutationBuilder(
+              "Singers", {"SingerId", "SingerInfo", "SingerInfoArray"})
+              .EmplaceRow(2, singer_info,
+                          std::vector<SingerInfoMessage>{singer_info})
+              .EmplaceRow(3, absl::optional<SingerInfoMessage>(),
+                          absl::optional<std::vector<SingerInfoMessage>>())
+              .Build()}});
+  for (auto& commit_result : commit_results) {
+    if (!commit_result) throw std::move(commit_result).status();
+  }
+  std::cout << "Update was successful "
+            << "[spanner_update_data_with_proto_message_column]\n";
```

Comment:
Here and below, is printing the `[spanner_*]` intentional?

Is there logic to when we print it and when we don't?

Reply:
Intentional, yes.  As you can see, many do, so I usually try to keep doing that (although now that we have `SampleBanner("tag")`, it is probably less important).  So, I'm afraid the only logic is: Is the "[tag]" even slightly helpful, and do you remember to add it.